File: Easy/No_0001_Two Sum/001.go
//Runtime: 8 ms, faster than 39.56% of Go online submissions for Two Sum.
//Memory Usage: 4 MB, less than 7.69% of Go online submissions for Two Sum.


func twoSum(nums []int, target int) []int {
    
    hash := make(map[int]int)
    for i, v := range nums {
        _, ok := hash[v]
        if ok == true{
            return []int{hash[v], i}
        } else{
            hash[target - v] = i
        }
    }
    return []int{0,0}
}

// 最後のreturn文は恐らく決まりみたいなもの？
// 使いたくない変数は_にする。（ohterwise it will throw error）

File: Easy/No_0001_Two Sum/001.rb

# Runtime: 60 ms, faster than 47.28% of Ruby online submissions for Two Sum.
# Memory Usage: 10.2 MB, less than 6.83% of Ruby online submissions for Two Sum.

def two_sum(nums, target)
    hash = {}
    
    nums.each_with_index do |n, i|
        if hash[target - n] != nil
            return hash[target - n], i
        end
        hash[n] = i
    end
end
